lms/services/canvas_api.py

```diff
@@ -91,10 +94,25 @@ def public_url(self, file_id):
 
         :rtype: str
         """
-        return self._helper.validated_response(
-            self._helper.public_url_request(self._oauth2_token.access_token, file_id),
+        oauth2_token = self._oauth2_token
+
+        return self.send_with_refresh_and_retry(
+            self._helper.public_url_request(oauth2_token.access_token, file_id),
             CanvasPublicURLResponseSchema,
-        ).parsed_params["public_url"]
+            oauth2_token.refresh_token,
+        )["public_url"]
```

Comment:
This is just the exact same request and response, but now sent via the new `send_with_refresh_and_retry()`